features/machine/machine.feature

```diff
@@ -0,0 +1,94 @@
+  # @author jhou@redhat.com
+  # @case_id OCP-22115
+  @smoke
+  Scenario: machine-api clusteroperator should be in Available state
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Available')` is stored in the :co_available clipboard
+    Then the expression should be true> cb.co_available["status"]=="True"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Degraded')` is stored in the :co_degraded clipboard
+    Then the expression should be true> cb.co_degraded["status"]=="False"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Upgradeable')` is stored in the :co_upgradable clipboard
+    Then the expression should be true> cb.co_upgradable["status"]=="True"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Progressing')` is stored in the :co_progressing clipboard
+    Then the expression should be true> cb.co_progressing["status"]=="False"
+
+  # @author jhou@redhat.com
+  # @case_id OCP-25436
+  @admin
+  @destructive
+  Scenario: Scale up and scale down a machineSet
+    Given I have an IPI deployment
+    And I store all machinesets to the :machinesets clipboard
+
+    Given evaluation of `cb.machinesets[0]` is stored in the :machineset clipboard
+    And evaluation of `cb.machineset.desired_replicas` is stored in the :replicas_original clipboard
+    And evaluation of `cb.machineset.desired_replicas.to_i + 1` is stored in the :replicas_expected clipboard
+    And I store the number of machines in the :num_machines_original clipboard
+
+    # scale up
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= cb.machineset.name %>   |
+      | replicas | <%= cb.replicas_expected %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+
+    # register clean-up just in case of failure
+    Given I register clean-up steps:
+    """
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= cb.machineset.name %>   |
+      | replicas | <%= cb.replicas_original %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+    """
+
+    # num of machines should increase
+    And I wait for the steps to pass:
+    """
+    Given I store the number of machines in the :num_machines_current clipboard
+    Then the expression should be true> cb.num_machines_current.to_i == cb.num_machines_original.to_i + 1
+    """
+
+    Given I store the provisioned machine in the :new_machine clipboard
+    And I wait for the node of machine named "<%= cb.new_machine %>" to appear
+
+    # new node should be ready
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | node               |
+      | resource_name | <%= cb.new_node %> |
+    Then the output should contain:
+      | Ready |
+    """
```

Comment:
can we use the following step?
```
^(I|admin) waits? for the#{OPT_QUOTED} (\w+) to become ready(?: in the#{OPT_QUOTED} project)?(?: up to (\d+) seconds)?$/
```

Reply:
I tried with `And admin waits for the "<%= cb.new_node %>" node to become ready up to 600 seconds`. But I got 

```
    And admin waits for the "<%= cb.new_node %>" node to become ready up to 600 seconds                       # features/step_definitions/resources.rb:114
      undefined method `[]' for false:FalseClass (NoMethodError)
      /Users/houjianwei/Projects/verification-tests/lib/openshift/resource.rb:336:in `block in wait_till_ready'
      /Users/houjianwei/Projects/verification-tests/lib/base_helper.rb:153:in `wait_for'
      /Users/houjianwei/Projects/verification-tests/lib/openshift/resource.rb:332:in `wait_till_ready'
      /Users/houjianwei/Projects/verification-tests/features/step_definitions/resources.rb:121:in `/^(I|admin) waits? for the(?: "(.+?)")? (\w+) to become ready(?: in the(?: "(.+?)")? project)?(?: up to (\d+) seconds)?$/'
      features/machine/machine.feature:70:in `And admin waits for the "<%= cb.new_node %>" node to become ready up to 600 seconds'
```

Probably because https://github.com/openshift/verification-tests/blob/master/lib/openshift/node.rb#L88 implemented a method `Ready?` that returns boolean.

Reply:
You're right, this is not ideal. I guess we can change `node` to also return a result. like:
```
    def ready?(user: nil, cached: true, quiet: false)
      status = get_cached_prop(prop: :status, user:user, cached: cached, quiet: quiet)
      result = {}
      result[:success] = status['conditions'].any? do |con|
        con['type'] == "Ready" && con['status'] == "True"
      end
      return result
    end
```

Reply:
Thank you, I've made the update.